src/google_form_to_engagement_db/configuration.py

```diff
@@ -67,10 +71,18 @@ def __init__(self, form_id, question_configurations, participant_id_configuratio
                                                If set, the participant uuid will be derived from the answer to an
                                                id question, otherwise it will be set to the form response id.
         :type participant_id_configuration: ParticipantIdConfiguration | None
+        ignore_invalid_mobile_numbers: bool = False
+        ignore_invalid_mobile_numbers: Whether to ignore invalid mobile numbers during validation.
         """
         self.form_id = form_id
         self.question_configurations = question_configurations
         self.participant_id_configuration = participant_id_configuration
+        self.ignore_invalid_mobile_numbers = ignore_invalid_mobile_numbers
+
+        if participant_id_configuration is not None and participant_id_configuration.id_type not in \
+            [GoogleFormParticipantIdTypes.KENYA_MOBILE_NUMBER]:
+            assert ignore_invalid_mobile_numbers == False, f"`ignore_invalid_mobile_numbers` cannot be set to True " \
```

Comment:
Does this mean if you want to use mobile numbers then invalid ones can never be ignored?

Reply:
No, it means invalid mobile numbers can never be ignored if we are not expecting a mobile number for example if we are expecting an email address `ignore_invalid_mobile_numbers` must be set to `False`.